### ISSUE TITLE
Indentation checks (LT04) don't work properly with leading commas (LT02)

### ISSUE THREAD

**Issue by daczarne** [sqlfluff/sqlfluff#6096]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using leading commas, indentation gets flagged as wrong even when it is correct.

### Expected Behaviour

For indentation to not be flagged as incorrect when it is correct.

### Observed Behaviour

Take the following query:

```sql
SELECT
  mt.id
  , mt.created_date
  , CASE
      WHEN mt.status = "Done" THEN 1
      WHEN mt.status = "Not done" THEN 2
      ELSE 3
    END AS status_value
FROM `my_project.my_dataset.my_table` AS mt
```

when running the lint command, it produces the following outcome:

![image](https://github.com/user-attachments/assets/91081099-6ce5-4ed9-9280-bdfa77fb2c3f)

### How to reproduce

Output is the result of running

```shell
sqlfluff lint --config .sqlfluff
```

on the provided query and with the provided config

### Dialect

BigQuery

### Version

SQLFluff version = 3.1.1
Python version = 3.10.1

### Configuration

```toml
# SQLFluff config
[sqlfluff]
dialect = bigquery
sql_file_exts = .sql
templater = jinja
encoding = utf-8
ignore_templated_areas = True

disable_noqa = False
warn_unused_ignores = True
fix_even_unparsable = False

large_file_skip_byte_limit = 0
large_file_skip_char_limit = 0
max_line_length = 120

processes = 1
nocolor = False
output_line_length = 80
runaway_limit = 10
verbose = 1

# *** LT02 *** #

[sqlfluff:rules:layout.indent]

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2

# *** LT04 *** #

[sqlfluff:rules:layout.commas]

[sqlfluff:layout:type:comma]
line_position = leading
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by apoorva-01** [comment]
Tried to repro this on current `main` (4.2.2) with the config from the issue, and I'm honestly not sure it's a bug. Wanted to check the intended behavior before touching anything.

With `tab_space_size = 2` and `line_position = leading`, `sqlfluff fix` settles on this, and it re-lints clean:

```sql
SELECT
  mt.id
  , mt.created_date
  , CASE
    WHEN mt.status = "Done" THEN 1
    WHEN mt.status = "Not done" THEN 2
    ELSE 3
  END AS status_value
FROM `my_project.my_dataset.my_table` AS mt
```

So it wants the `WHEN` lines at 4 spaces and `END` at 2, where the original had 6 and 4.

I checked whether the leading comma was throwing off the indent and it doesn't look like it. Leading vs trailing commas give the exact same expected indent (tested with the CASE above and a plain `foo(...)` call), so the `, ` isn't being counted as indentation. The difference is that the reported version lines the body up visually under `CASE` (column 4, after the `, `), while sqlfluff indents by logical level.

So before I write anything- should sqlfluff support that visual/hanging alignment after a leading comma, or is the current behavior correct and this is working as intended?
